include/core/evaporation_model.h

```diff
@@ -0,0 +1,683 @@
+  {
+    const unsigned int         n_pts = momentum_flux_vector.size();
+    const std::vector<double> &temperature =
+      field_vectors.at(field::temperature);
+
+    std::vector<double> saturation_pressure_vector(n_pts);
+    vector_saturation_pressure(field_vectors, saturation_pressure_vector);
+
+    std::vector<double> mass_flux_vector(n_pts);
+    vector_mass_flux(field_vectors, mass_flux_vector);
+
+    for (unsigned int i = 0; i < momentum_flux_vector.size(); ++i)
+      {
+        const double temperature_inv = 1.0 / temperature[i];
+
+        const double vapor_saturation_density_value = molar_mass*
+          saturation_pressure_vector[i] * R_inv * temperature_inv;
+
+        // rho_vap = 0.31*rho_sat according to Anisimov and Khokhlov 1995
+        const double vapor_density_inv =
+          1.0 / (0.31 * vapor_saturation_density_value);
+
+        momentum_flux_vector[i] =
+          -mass_flux_vector[i] * mass_flux_vector[i] *
+            (liquid_density_inv - vapor_density_inv) +
+          recoil_pressure_coefficient * saturation_pressure_vector[i];
+      }
+  }
+
+  /**
+   * @brief momentum_flux_jacobian Calculates the jacobian (the partial derivative)
+   * of the evaporation recoil pressure with respect to a field
+   * @param field_values Value of the various fields on which the recoil pressure
+   * may depend.
+   * @param id Indicator of the field with respect to which the jacobian
+   * should be calculated
+   * @return value of the partial derivative of the recoil pressure with respect
+   * to the field.
+   */
+  double
+  momentum_flux_jacobian(const std::map<field, double> & /*field_values*/,
+                         const field /*id*/) override
+  {
+    return 0.0;
+  }
+
+  /**
+   * @brief vector_momentum_flux_jacobian Calculate the derivative of the
+   * evaporation recoil pressure with respect to a field
+   * @param field_vectors Vector for the values of the fields on which the recoil
+   * pressure may depend.
+   * @param id Identifier of the field with respect to which a derivative should
+   * be calculated
+   * @param jacobian Vector of the value of the derivative of the recoil pressure
+   * with respect to the field id
+   */
+  void
+  vector_momentum_flux_jacobian(
+    const std::map<field, std::vector<double>> & /*field_vectors*/,
+    const field /*id*/,
+    std::vector<double> &jacobian_vector) override
+  {
+    std::fill(jacobian_vector.begin(), jacobian_vector.end(), 0);
+  }
+
+private:
+  const double evaporation_coefficient;
+  const double recoil_pressure_coefficient;
+
+  const double molar_mass;
+  const double boiling_temperature_inv;
+  const double latent_heat_evaporation;
+  const double ambient_pressure;
+  const double ambient_gas_density_inv;
+  const double liquid_density_inv;
+
+  // 1/R
+  double R_inv;
+
+  // L_vap*M/R
+  double L_vap_x_M_x_R_inv;
+
+  // M/(2*PI*R)
+  double M_x_2PI_inv_x_R_inv;
```

Comment:
I don't think pre-storing these variables for computational efficiency is necessarily worth it, but if you believe so, we can keep it